### PR TITLE
tgis-standalone/caikit-standalone manifests

### PR DESCRIPTION
**Pull request by dtrifiro**
- add manifests for inference using only caikit (caikit-standalone)
- add readliness/liveness probes for tgis/caikit-tgis (partially solves https://github.com/opendatahub-io/caikit-tgis-serving/issues/170, https://github.com/opendatahub-io/caikit-tgis-serving/issues/156)
- split example manifests in:
    - `demo/kserve/custom-manifests/caikit/caikit-standalone` (cakit only)
    - `demo/kserve/custom-manifests/caikit/caikit-tgis` (caikit frontend with tgis backend)
    - `demo/kserve/custom-manifests/tgis` (tgis only)
- misc docs cleanups

### CODE REVIEW

File: demo/kserve/deploy-remove.md [hunk 1]
```diff
@@ -123,151 +97,151 @@ Note: The **flan-t5-small** LLM model has been containerized into an S3 MinIO bu
...
-            "logprob": -3.3622500896453857
-          }
-        ],
-        "details": {
-          "generated_tokens": 2
-        }
-      }
-      ....
-      ```
+   - http only. Perform inference with HTTP. This example uses cURL.
```
Comment (dtrifiro) [inline demo/kserve/deploy-remove.md:108]: Apart from this line, this block hasn't really changed, only the level of indentation has

File: docs/README.md [hunk 2]
```diff
@@ -1,45 +1,45 @@
...
+- **[Text Generation Inference Server (TGIS)](https://github.com/opendatahub-io/text-generation-inference)**: Runtime that loads the models and provides the inference engine.
+- **[KServe](https://github.com/opendatahub-io/kserve)**: A Kubernetes Custom Resource Definition that orchestrates model serving for all types of models. It includes serving runtimes that implement the loading of given types of model servers. KServe handles the lifecycle of the deployment object, storage access, and networking setup.
+- **[Service Mesh](https://docs.openshift.com/container-platform/4.13/service_mesh/v2x/ossm-architecture.html)** (istio): The service mesh networking layer that manages traffic flows and enforces access policies.
+- **[Serverless](https://docs.openshift.com/serverless/1.29/about/about-serverless.html)** (knative): A cloud-native development model that allows for serverless deployments of data models.
 
 ## Architecture of the stack
 
 ![KServe+Knative+Istio+Caikit_TGIS Diagram](https://github.com/opendatahub-io/caikit-tgis-serving/assets/8479010/7009b95d-0f6f-4f18-b0e6-355f360a5ad1)
```
Comment (dtrifiro) [inline docs/README.md:16]: @heyselbi I see you're the last person to have updated this diagram. Do you think we should update it once again to reflect the new architecture?

File: demo/kserve/custom-manifests/caikit/caikit-standalone/caikit-standalone-servingruntime-grpc.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  name: caikit-standalone-runtime-grpc
+spec:
+  multiModel: false
+  supportedModelFormats:
+    # Note: this currently *only* supports caikit format models
+    - autoSelect: true
+      name: caikit
+  containers:
+    - name: kserve-container
+      image: quay.io/opendatahub/caikit-nlp:stable
```
Comment (dtrifiro) [inline demo/kserve/custom-manifests/caikit/caikit-standalone/caikit-standalone-servingruntime-grpc.yaml:13]: This image isn't available (yet). Waiting on https://github.com/openshift/release/pull/46641 for the workflows required to push the image on quay and the creation of a release branch on https://github.com/opendatahub-io/caikit-nlp

**Comment by Xaenalt** [comment]
Overall lgtm, once https://github.com/openshift/release/pull/46641 merges then will be good to merge

**Comment by dtrifiro** [comment]
By further testing, there currently are some issues with readiness probes being removed by knative-serving and grpc liveness probes not working under certain circumstances. I'll update this soon with the proper warnings

File: demo/kserve/custom-manifests/tgis/tgis-servingruntime.yaml [hunk 4]
```diff
@@ -11,8 +11,8 @@ spec:
     - name: kserve-container
       image: quay.io/opendatahub/text-generation-inference:stable
       command: ["text-generation-launcher"]
-      args: 
-        - "--model-name=/mnt/models/"
+      args:
+        - "--model-name=/mnt/models/<model>"
```
Comment (bdattoma) [inline demo/kserve/custom-manifests/tgis/tgis-servingruntime.yaml:15]: I think we should remove `<model>`. I tried on my side, and it fails with it

**Pull request merged**